pallets/gear/src/weights.rs

```diff
@@ -2070,6 +2081,116 @@ impl<T: frame_system::Config> WeightInfo for SubstrateWeight<T> {
+        // Proof Size summary in bytes:
+        //  Measured:  `714`
+        //  Estimated: `16349`
+        // Minimum execution time: 19_000_000 picoseconds.
+        Weight::from_parts(21_000_000, 16349)
+            .saturating_add(T::DbWeight::get().reads(5_u64))
+            .saturating_add(T::DbWeight::get().writes(4_u64))
+    }
+    fn tasks_send_user_message_to_mailbox() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `734`
+        //  Estimated: `21234`
+        // Minimum execution time: 18_000_000 picoseconds.
+        Weight::from_parts(19_000_000, 21234)
+            .saturating_add(T::DbWeight::get().reads(6_u64))
+            .saturating_add(T::DbWeight::get().writes(5_u64))
+    }
+    fn tasks_send_user_message() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `617`
+        //  Estimated: `25736`
+        // Minimum execution time: 20_000_000 picoseconds.
+        Weight::from_parts(21_000_000, 25736)
+            .saturating_add(T::DbWeight::get().reads(9_u64))
+            .saturating_add(T::DbWeight::get().writes(8_u64))
+    }
+    fn tasks_send_dispatch() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `561`
+        //  Estimated: `19705`
+        // Minimum execution time: 15_000_000 picoseconds.
+        Weight::from_parts(16_000_000, 19705)
+            .saturating_add(T::DbWeight::get().reads(7_u64))
+            .saturating_add(T::DbWeight::get().writes(6_u64))
+    }
+    fn tasks_wake_message() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `823`
+        //  Estimated: `25565`
+        // Minimum execution time: 20_000_000 picoseconds.
+        Weight::from_parts(21_000_000, 25565)
+            .saturating_add(T::DbWeight::get().reads(8_u64))
+            .saturating_add(T::DbWeight::get().writes(6_u64))
+    }
+    fn tasks_wake_message_no_wake() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `80`
+        //  Estimated: `3545`
+        // Minimum execution time: 2_000_000 picoseconds.
+        Weight::from_parts(2_000_000, 3545)
+            .saturating_add(T::DbWeight::get().reads(1_u64))
+    }
+    fn tasks_remove_from_waitlist() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `1200`
+        //  Estimated: `47840`
+        // Minimum execution time: 41_000_000 picoseconds.
+        Weight::from_parts(44_000_000, 47840)
+            .saturating_add(T::DbWeight::get().reads(14_u64))
+            .saturating_add(T::DbWeight::get().writes(11_u64))
+    }
+    fn tasks_remove_from_mailbox() -> Weight {
+        // Proof Size summary in bytes:
+        //  Measured:  `907`
+        //  Estimated: `37005`
+        // Minimum execution time: 30_000_000 picoseconds.
+        Weight::from_parts(33_000_000, 37005)
+            .saturating_add(T::DbWeight::get().reads(12_u64))
+            .saturating_add(T::DbWeight::get().writes(11_u64))
```

Comment:
Unexpectedly big weight for some tasks. I think it's because of gas-tree. How you can guaranty that this weight is constant and does not depends on gas-tree current state?

(Also I have question now, why do we really need to create gas tree node in remove from mail box).

Reply:
on removal from mailbox goes auto reply sending that requires gas node

Reply:
> on removal from mailbox goes auto reply sending that requires gas node
Thank you! But what shell we do with cost dependency from gas tree state. Am I right that this can cause different amount of read/write to storage?